client/asset/btc/btc.go

```diff
@@ -821,55 +822,76 @@ func (btc *ExchangeWallet) FundingCoins(ids []dex.Bytes) (asset.Coins, error) {
 		pt := newOutPoint(txHash, vout)
 		fundingCoin, found := btc.fundingCoins[pt]
 		if found {
-			coins = append(coins, newOutput(btc.node, txHash, vout, fundingCoin.amount, fundingCoin.redeemScript))
+			coins = append(coins, newOutput(btc.node, txHash, vout, fundingCoin.amount))
 			continue
 		}
-		notFound[pt] = struct{}{}
+		notFound[pt] = true
 	}
-
 	if len(notFound) == 0 {
 		return coins, nil
 	}
-	_, utxoMap, _, err := btc.spendableUTXOs(0)
+
+	// Check locked outputs for not found coins.
+	lockedOutpoints, err := btc.wallet.ListLockUnspent()
 	if err != nil {
 		return nil, err
 	}
-
-	lockers := make([]*output, 0, len(ids))
-
-	for _, id := range ids {
-		txHash, vout, err := decodeCoinID(id)
+	for _, rpcOP := range lockedOutpoints {
+		txHash, err := chainhash.NewHashFromStr(rpcOP.TxID)
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("error decoding txid from rpc server %s: %v", rpcOP.TxID, err)
 		}
-		pt := newOutPoint(txHash, vout)
+		pt := newOutPoint(txHash, rpcOP.Vout)
+		if notFound[pt] {
+			txOut, err := btc.node.GetTxOut(txHash, rpcOP.Vout, true)
+			if err != nil {
+				return nil, fmt.Errorf("gettxout error for locked outpoint %v: %v", pt.String(), err)
+			}
+			var address string
+			if len(txOut.ScriptPubKey.Addresses) > 0 {
+				address = txOut.ScriptPubKey.Addresses[0]
+			}
+			utxo := &utxo{
+				txHash:  txHash,
+				vout:    rpcOP.Vout,
+				address: address,
+				amount:  toSatoshi(txOut.Value),
+			}
+			coin := newOutput(btc.node, txHash, rpcOP.Vout, toSatoshi(txOut.Value))
+			coins = append(coins, coin)
+			btc.fundingCoins[pt] = utxo
+			delete(notFound, pt)
+			if len(notFound) == 0 {
+				return coins, nil
+			}
+		}
+	}
+
+	// Some funding coins still not found after checking locked outputs.
+	// Check wallet unspent outputs as last resort. Lock the coins if
+	// they're found.
+	_, utxoMap, _, err := btc.spendableUTXOs(0)
+	if err != nil {
+		return nil, err
+	}
+	unlockedOutputs := make([]*output, 0, len(ids))
+	for pt := range notFound {
 		utxo, found := utxoMap[pt]
 		if !found {
 			return nil, fmt.Errorf("funding coin %s not found", pt.String())
 		}
-		btc.fundingCoins[pt] = utxo
-		coin := newOutput(btc.node, utxo.txHash, utxo.vout, utxo.amount, utxo.redeemScript)
+		btc.fundingCoins[pt] = utxo.utxo
+		coin := newOutput(btc.node, utxo.txHash, utxo.vout, utxo.amount)
 		coins = append(coins, coin)
-		lockers = append(lockers, coin)
+		unlockedOutputs = append(unlockedOutputs, coin)
 		delete(notFound, pt)
-		if len(notFound) == 0 {
-			break
-		}
-	}
-	if len(notFound) != 0 {
-		ids := make([]string, 0, len(notFound))
-		for pt := range notFound {
-			ids = append(ids, pt.String())
-		}
-		return nil, fmt.Errorf("coins not found: %s", strings.Join(ids, ", "))
 	}
-	err = btc.wallet.LockUnspent(false, lockers)
+	btc.log.Debugf("locking funding coins that were unlocked %v", unlockedOutputs)
```

Comment:
Please log with Capital sentence.  It's obv not important, but our logging should be as consistent as possible.